### PR TITLE
Custom web driver provider set up, import/export integration test added, maven webdriver download plugin included

### PR DESCRIPTION
**Pull request by mcada**
Please look at these changes and lets discuss what should I change or if it is suitable for our purposes, there are still some TODOs which I have to figure out...

### CODE REVIEW

File: ui-tests/src/test/java/io/syndesis/qe/CustomWebDriverProvider.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Hashtable;
+import java.util.Map;
+
+@Slf4j
+public class CustomWebDriverProvider implements WebDriverProvider {
+    public static final String downloadFilePath = System.getProperty("user.dir") + File.separator + "target";
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/CustomWebDriverProvider.java]: Hmm, I would go for something like `tmp/download` inside module dir. Also you can achieve this with `Paths.get("tmp/download").toAbsolute()`

File: ui-tests/pom.xml [hunk 2]
```diff
@@ -1,53 +1,79 @@
...
+        </dependency>
 
-	</dependencies>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.github.webdriverextensions</groupId>
+                <artifactId>webdriverextensions-maven-plugin</artifactId>
+                <version>3.1.3</version>
```
Comment (dsimansk) [inline ui-tests/pom.xml]: Version to maven property and property to parent pom, pls.

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java [hunk 3]
```diff
@@ -320,4 +323,31 @@ public void setJmsPublishData(DataTable sourceMappingData) {
...
+	public void startIntegration(String integrationName) {
+		detailPage.getButton("Start Integration").shouldBe(visible).click();
+
+		ModalDialogPage modal = new ModalDialogPage();
+		modal.getButton("OK").shouldBe(visible).click();
+	}
+
+	@And("^Wait until there is no integration pod with name \"([^\"]*)\"$")
+	public void waitForIntegrationPodShutdown(String integartionPodName) throws InterruptedException {
+		while(!OpenShiftWaitUtils.areNoPodsPresent(integartionPodName).getAsBoolean()) {
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: Good approach, but lets reiterate :)
```
OpenShiftWaitUtils.assertEventually("Still running", OpenShiftWaitUtils.areNoPodsPresent(), 1000, 5 * 60 *1000);
```

File: ui-tests/src/test/java/io/syndesis/qe/CustomWebDriverProvider.java (hunk 1 above)
Comment (mcada) [reply ui-tests/src/test/java/io/syndesis/qe/CustomWebDriverProvider.java]: This property is used for integration export and all other downloads from UI tests - it should be somewhere in target dir so it is deleted with every mvn clean. What about this: 
Paths.get("target" + File.separator + "download").toAbsolutePath().toString();

Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/CustomWebDriverProvider.java]: Ofc, separator works for me 👍 

I'd like to skip target dir and rather add & configure maven-clean-plugin to do it's job <module-dir>/tmp :)

**Comment by mcada** [comment]
Updated, I guess it is no more WIP as it works well but it should be tested on another machines and on MACs.

File: ui-tests/pom.xml [hunk 4]
```diff
@@ -1,53 +1,92 @@
...
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <artifactId>maven-clean-plugin</artifactId>
+                <version>${maven.clean.plugin.version}</version>
+                <configuration>
+                    <filesets>
+                        <fileset>
+                            <directory>tmp</directory>
```
Comment (dsimansk) [inline ui-tests/pom.xml]: I would add to keep the tmp directory itself
```
<includes>
      <include>**/*</include>
</includes>```

**Comment by dsimansk** [comment]
@mcada pls update with your latest changes && rebase on master and imho it's good to go. Other improvements can be done later.

**Comment by mcada** [comment]
done, working on both firefox and chrome :)

**Comment by dsimansk** [comment]
@mcada ok, remove WIP label once you're done. 🦄

**Comment by mcada** [comment]
done

**Pull request merged**